linkerd/app/inbound/src/policy/http/tests.rs

```diff
@@ -0,0 +1,230 @@
+use super::*;
+use crate::policy::{Authentication, Authorization, Meta, Protocol, ServerPolicy};
+use linkerd_app_core::{svc::Service, Infallible};
+use std::sync::Arc;
+
+macro_rules! new_svc {
+    ($proto:expr, $conn:expr) => {{
+        let (policy, tx) = AllowPolicy::for_test(
+            $conn.dst,
+            ServerPolicy {
+                protocol: $proto,
+                meta: Arc::new(Meta::Resource {
+                    group: "policy.linkerd.io".into(),
+                    kind: "server".into(),
+                    name: "testsrv".into(),
+                }),
+            },
+        );
+        let svc = HttpPolicyService {
+            target: (),
+            policy,
+            connection: $conn,
+            metrics: HttpAuthzMetrics::default(),
+            inner: |(permit, _): (HttpRoutePermit, ())| {
+                svc::mk(move |_: http::Request<hyper::Body>| {
+                    let permit = permit.clone();
+                    async move {
+                        let mut rsp = http::Response::builder()
+                            .body(hyper::Body::default())
+                            .unwrap();
+                        rsp.extensions_mut().insert(permit.clone());
+                        Ok::<_, Infallible>(rsp)
+                    }
+                })
+            },
+        };
+        (svc, tx)
+    }};
+
+    ($proto:expr) => {{
+        let conn = ConnectionMeta {
+            dst: OrigDstAddr(([192, 168, 3, 4], 8080).into()),
+            client: Remote(ClientAddr(([192, 168, 3, 3], 30120).into())),
+            tls: tls::ConditionalServerTls::Some(tls::ServerTls::Established {
+                client_id: Some("foo.bar.bah".parse().unwrap()),
+                negotiated_protocol: None,
+            }),
+        };
+        new_svc!($proto, conn)
+    }};
+}
```

Comment:
nit, TIOLI: this kind of feels like it _could_ just be a pair of functions rather than a macro, but i'm not sure if it's really a big deal either way...

Reply:
Yeah, but I don't want to have to name the response type.

Reply:
oh, missed that. yeah, let's keep the macro then!